### PR TITLE
basics for OSS::Badge component

### PR DESCRIPTION
**Pull request by phndiaye**
### What does this PR do?

New component for using Badges.

### What are the observable changes?

Screenshot 2021-12-13 at 15 02 56

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/badge.hbs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class={{this.computedClass}} ...attributes>
+  {{#if @icon}}
+    <i class="{{@icon}} text-color-bright-purple"></i>
+  {{else if @image}}
+      {{!-- template-lint-disable require-valid-alt-text  --}}
```
Comment (Miexil) [inline addon/components/o-s-s/badge.hbs:5]: Or, you know, put a default alt-text ? 😇

File: addon/components/o-s-s/badge.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  text: string;
+  size: SizeType;
+}
+
+export default class OSSBadge extends Component<OSSBadgeArgs> {
+  constructor(owner: unknown, args: OSSBadgeArgs) {
+    super(owner, args);
+
+    const contentArguments = [args.icon, args.image, args.text].filter((arg: string) => arg);
+    console.log(contentArguments);
```
Comment (Miexil) [inline addon/components/o-s-s/badge.ts:23]: Forgotten log

File: app/styles/oss-components.less [hunk 3]
```diff
@@ -1,8 +1,34 @@
...
+@import 'power-select';
+@import 'toast';
+@import 'button';
+
+.upf-badge {
+  .border;
+  .border-color-default;
+  /*.fx-row;*/
+  /*.fx-malign-center;*/
+  /*.fx-xalign-center;*/
```
Comment (Miexil) [inline app/styles/oss-components.less:15]: comments :-)

File: addon/components/o-s-s/badge.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (contentArguments.length === 0 || contentArguments.length > 1) {
```
Comment (JulienVannier66) [inline addon/components/o-s-s/badge.ts:25]:
```suggestion
    if (contentArguments.length !== 1) {
```

**Review by Miexil — APPROVED** [review]

**Review by JulienVannier66 — APPROVED** [review]

**Pull request merged**